hbase-oss/src/main/java/org/apache/hadoop/hbase/oss/sync/ZKTreeLockManager.java

```diff
@@ -161,7 +161,7 @@ protected void writeUnlock(Path p) throws IOException {
       get(p).writeLock().release();
```

Comment:
This 'get' call creates the znode if it's not in the cache, so call to writeUnlock after deleting the znode would create it again.

Reply:
Yeah, right as usual. I want to figure out why my unit test isn't failing like I think it should, given this.

However, I think I can interpret you see this as an improvement, but there is more work to be done (citing the problems, but approving this PR)? Just wanted to double check.